oak_crypto/src/noise_nk/rustcrypto.rs

```diff
@@ -0,0 +1,212 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! This was copied from Chromium's third_party/cloud_authenticator, which has
+//! compatible copyright and ownership (Apache 2.0, Google).
+//!
+//! The original code supported multiple back-end crypto libraries, but was
+//! heavily influenced by the first: Ring.  The result was a bit hackish, and
+//! should be cleaned up.
```

Comment:
It looks like this file only contains wrappers that call very similarly named functions. If we only plan to use RustCrypto in Oak, could we use the corresponding primitives directly without wrappers?

Reply:
Let's figure that out in a follow-on PR.  What is clear is that this set of APIs is not much like either BoringSSL or Rust crypto.  There is value in having the shim so users can use BoringSSL if needed, but it may make more sense to do this at a level where all Oak crypto can use it?

This is complex enough I prefer to kick it down the road.  Let's figure it out in a separate PR.